### DISCUSSION TITLE
Install on Shared Hosting

### DISCUSSION THREAD

**Discussion by modante** [chillerlan/php-qrcode#109] [Q&A]
Hello. I have a shared hosting where I am not able to use ssh in order to run composer. The typical CPanel hosting. I have run some php scripts in it and works great but I don't know how to make working php-qrcode in it.
I have installed on my local machine with composer and I have copied all the "vendor" dir on the root of the remote server but I am lost with the namespaces `namespace chillerlan\QRCodeExamples;` and the autoload.php `require_once __DIR__.'/../vendor/autoload.php';`
I cannot get to work svg.php in order to add a qrcode to an existing svg graphic.
If anybody knows how can I accomplish this task, would be great :-D

**Comment by modante** [comment]
Sorry. I think this is not the place to ask this question, would be better in Discussions. Doesn't it?
I can delete it and create a Discussion if you think so.

**Comment by codemasher** [comment]
The examples will only work if you installed the package with dev dependencies (`require-dev`). It *should* work in your host environment if you installed it locally that way, however it is not recommended outside of development.

**Comment by modante** [comment]
Excuse my inexperience but I tried to install php-qrcode with require-dev typing `composer require --dev chillerlan/php-qrcode` and I got this:
```
$ composer require --dev chillerlan/php-qrcode
Using version ^3.0 for chillerlan/php-qrcode
chillerlan/php-qrcode is currently present in the require key and you ran the command with the --dev flag, which would move it to the require-dev key.
Do you want to move this requirement? [no]? yes
./composer.json has been updated
Running composer update chillerlan/php-qrcode
Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Writing lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Generating autoload files
1 package you are using is looking for funding.
Use the `composer fund` command to find out more!
```
Then I copied the complete vendor dir (that includes chillerlan dir, composer dir and autoload.php) on shared hosting and opened the complete path to svg.php and I get nothing.
Taking a look at the code at the begining of svg.php, says: `require_once __DIR__.'/../vendor/autoload.php';`
but the autoload.php is 3 dirs upper examples dir, then I think is not loading anything.

My dir structure is:
- vendor
     - chillerlan
          - php-qrcode
               - examples
               - public
               - src
               - tests
          - php-settings-container
     - composer

Should I change anything in order to make it work?
Thank you very much for your support

**Reply by codemasher** [reply]
Ah, my bad. The include from the example directory will only work when the project is cloned, in which case the `vendor` directory would be in the `php-qrcode` root. I assume you don't see an error because error reporting is turned off on your remote host. However, the autoloader and namespaces should work properly, so you'd just need to copy the `svg.php` to your project, change/remove the namespace and point to the correct path to `autoload.php`. In that case you also wouln't need the dev namespaces, which are used in the extended output examples only.

**Reply by modante** [reply]
Great!!!! Now it is working like a charm!!! It was easiest than I thought :-D
Thank you very much :-)

**Reply by codemasher** [reply]
That's great to hear! Cheers!
